### DISCUSSION TITLE
Hash autocompletion

### DISCUSSION THREAD

**Discussion by azaslavsky** [arxanas/git-branchless#895] [Ideas]
I'm thinking back to my time at a company that used a similar solution to git-branchless, and one nice feature we had was hash autocompletion. It essentially looked at all the hashes printed by `git smartlog`, and for commands like `git move` that require naming specific commits, it would provide an autocomplete.  For example, if you have commits identified by hashes `aaaaaaa` and `bbbbbbb`, typing `git move -d a<TAB>` would result in an autocompletion populating to `git move -d aaaaaaa`.  I know the project has generally avoided first-class autocomplete so far, but perhaps there is a script that could scrape the smartlog output in some clever way to provide autocomplete support?

**Comment by arxanas** [comment]
A user created a script for autocompletions in zsh here, you could try that: https://github.com/arxanas/git-branchless/discussions/217#discussioncomment-4875866
